From aa32d42e3551fcd406b24bcd1f3d16776ba1bccb Mon Sep 17 00:00:00 2001
From: Xishi Qiu <qiuxishi@huawei.com>
Date: Fri, 3 Dec 2021 18:16:06 +0800
Subject: [PATCH] pci: Enable acs for QLogic HBA cards
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: aa32d42e3551fcd406b24bcd1f3d16776ba1bccb


euler inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4K05G?from=project-issue
CVE: N/A

-------------------------------------------------

Add support of port isolation for QLogic HBA cards.

Signed-off-by: Xishi Qiu <qiuxishi@huawei.com>
Signed-off-by: Fang Ying <fangying1@huawei.com>
Signed-off-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Hui Wang <john.wanghui@huawei.com>
Signed-off-by: Zhang Xiaoxu <zhangxiaoxu5@huawei.com>

Confilicts:
	drivers/pci/quirks.c

Signed-off-by: Xuefeng Wang <wxf.wang@hisilicon.com>
Reviewed-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Jialin Zhang <zhangjialin11@huawei.com>

Confilicts:
        drivers/pci/quirks.c
Reviewed-by: wangxiongfeng <wangxiongfeng2@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/pci/quirks.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index c5d38cb329d7..96245b93e613 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -4889,6 +4889,8 @@ static const struct pci_dev_acs_enabled {
 	{ PCI_VENDOR_ID_INTEL, PCI_ANY_ID, pci_quirk_intel_spt_pch_acs },
 	{ 0x19a2, 0x710, pci_quirk_mf_endpoint_acs }, /* Emulex BE3-R */
 	{ 0x10df, 0x720, pci_quirk_mf_endpoint_acs }, /* Emulex Skyhawk-R */
+	{ 0x1077, 0x2031, pci_quirk_mf_endpoint_acs}, /* QLogic QL2672 */
+	{ 0x1077, 0x2532, pci_quirk_mf_endpoint_acs},
 	/* Cavium ThunderX */
 	{ PCI_VENDOR_ID_CAVIUM, PCI_ANY_ID, pci_quirk_cavium_acs },
 	/* Cavium multi-function devices */
-- 
2.26.2

